RdlMigration/ConvertRDL.cs

```diff
@@ -101,8 +101,11 @@ private void ConvertAndUploadReport(PowerBIClientWrapper powerBIClient, RdlFileI
                     XElement[] dataSets = rdlFileIO.GetDataSets(reportPath, out Dictionary<string, XElement> referenceDataSetMap);
                     DataSource[] dataSources = rdlFileIO.GetDataSources(reportPath);
 
-                    var convertedFile = ConvertFile(report, dataSources, referenceDataSetMap);
-                    SaveAndCopyStream(reportName, convertedFile, $"output\\{reportName}_convert.rdl");
+                    // If two or more datasets reference the same datasource, references with duplicated names will appear datasources.
+                    var uniqDatasources = dataSources.ToList().Select(ds => ds.Name).Distinct().Select(n => dataSources.ToList().First(ds => ds.Name.Equals(n)));
```

Comment:
You should not need the first "ToList()" since all the operations will work on the Enumerable